aws/service.go

```diff
@@ -1763,6 +1764,15 @@ func getSessionWithMaxRetries(ctx context.Context, d *plugin.QueryData, region s
 	// get aws config info
 	awsConfig := GetConfig(d.Connection)
 
+	// Configure the maximum number number of retry user wants to.
+	if awsConfig.MaxRetryCount != nil && *awsConfig.MaxRetryCount != "" {
```

Comment:
I think we should check this in the `getSession` method instead, as we pass in `9` there hardcoded, but it feels like it's better to keep the config check/defaulting to 9 all in one place.